src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.StackRowStack.cs

```diff
@@ -11,18 +11,27 @@ public sealed partial class JsonDocument
     {
         private struct StackRowStack : IDisposable
         {
-            private byte[] _rentedBuffer;
+            private byte[]? _rentedBuffer;
             private int _topOfStack;
 
             internal StackRowStack(int initialSize)
             {
-                _rentedBuffer = ArrayPool<byte>.Shared.Rent(initialSize);
-                _topOfStack = _rentedBuffer.Length;
+                if (initialSize == -1)
```

Comment:
Any noticeable perf/allocation regression to the existing `JsonDocument.Parse` methods?

Reply:
No; I'll provide a summary in a bit.